Explorer/Assets/DCL/WebRequests/Analytics/WebRequestsContainer.cs

```diff
@@ -1,24 +1,57 @@
+        public IWebRequestController WebRequestController { get; }
 
-        public IWebRequestsAnalyticsContainer AnalyticsContainer { get; private set; }
+        public IWebRequestsAnalyticsContainer AnalyticsContainer { get; }
 
-        public static WebRequestsContainer Create(IWeb3IdentityCache web3IdentityProvider)
+        private WebRequestsContainer(IWebRequestController webRequestController, IWebRequestsAnalyticsContainer analyticsContainer)
         {
-            var analyticsContainer = new WebRequestsAnalyticsContainer().AddTrackedMetric<ActiveCounter>()
-                                                                        .AddTrackedMetric<Total>()
-                                                                        .AddTrackedMetric<TotalFailed>()
-                                                                        .AddTrackedMetric<BandwidthDown>()
-                                                                        .AddTrackedMetric<BandwidthUp>();
-
-            var webRequestController = new LogWebRequestController(new WebRequestController(analyticsContainer, web3IdentityProvider));
-            return new WebRequestsContainer { WebRequestController = webRequestController, AnalyticsContainer = analyticsContainer };
+            WebRequestController = webRequestController;
+            AnalyticsContainer = analyticsContainer;
         }
+
+        public static WebRequestsContainer Create(IWeb3IdentityCache web3IdentityProvider, WebRequestsContainerParams containerParams)
+        {
+            var analyticsContainer = new WebRequestsAnalyticsContainer()
+                                    .AddTrackedMetric<ActiveCounter>()
+                                    .AddTrackedMetric<Total>()
+                                    .AddTrackedMetric<TotalFailed>()
+                                    .AddTrackedMetric<BandwidthDown>()
+                                    .AddTrackedMetric<BandwidthUp>();
+
+            IWebRequestController webRequestController = new LogWebRequestController(
+                new WebRequestController(analyticsContainer, web3IdentityProvider)
+            );
+
+            if (containerParams.ShouldApplyDelay())
+            {
+                ReportHub.Log(ReportCategory.WEB_REQUESTS, "Artificial delay enabled");
+                webRequestController = new ArtificialDelayWebRequestController(webRequestController, containerParams.ArtificialDelaySeconds);
+            }
+
+            return new WebRequestsContainer(webRequestController, analyticsContainer);
+        }
+    }
+
+    [Serializable]
+    public class WebRequestsContainerParams
+    {
+        [SerializeField] private bool useDelay = true;
+        [SerializeField] private float artificialDelaySeconds = 10;
+
+        public float ArtificialDelaySeconds => artificialDelaySeconds;
+
+        public bool ShouldApplyDelay() =>
+            Application.isEditor
```

Comment:
Maybe in the future we want to use the delay even in the build, it could be useful to expose the delay in the debug panel and remove the isEditor check

Reply:
good idea, I'm on it